File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    myDict = {}
    for i in range(len(want)):
        myDict[want[i]] = number[i]
        
    for i in range(len(discount)):
        if i + 10 <= len(discount):
            tmp = discount[i:i+10]
            discountDict = {}
            for item in tmp:
                try:
                    discountDict[item] += 1
                except:
                    discountDict[item] = 1
            
            keys = list(discountDict)
            isOk = True
            
            for key in keys:
                try:
                    if myDict[key] != discountDict[key]:
                        isOk = False
                except:
                    isOk = False
                    
            if isOk:
                answer += 1
                
    
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    
    unordered_map<string, int> TargetMap;
    unordered_map<string, int> NowMap;

    for(int i = 0; i < number.size(); i++){
        TargetMap[want[i]] = number[i];
    }
    
    for(int i = 0; i < 10; i++){
        if(NowMap[discount[i]]){
            NowMap[discount[i]] += 1;
        }
        else{
            NowMap[discount[i]] = 1;
        }    
    }
    
    bool isOk = true;
    for(int i = 0; i < want.size(); i++){
        if(TargetMap[want[i]] != NowMap[want[i]]){
            isOk = false;
            break;
        }
    }
    
    if(isOk){
        answer += 1;
    }
    
    for(int i = 10; i < discount.size(); i++){
        NowMap[discount[i - 10]] -= 1;
        NowMap[discount[i]] += 1;
        
        bool isOk = true;
        for(int i = 0; i < want.size(); i++){
            if(TargetMap[want[i]] != NowMap[want[i]]){
                isOk = false;
                break;
            }
        }

        if(isOk){
            answer += 1;
        }

    }
    
    
    return answer;
}